go_modules/lib/dependabot/go_modules/resolvability_errors.rb

```diff
@@ -6,7 +6,19 @@ module ResolvabilityErrors
       GITHUB_REPO_REGEX = %r{github.com/[^:@]*}.freeze
 
       def self.handle(message, credentials:, goprivate:)
+        # TODO: currently this matches last. Instead, if more than one match, and they
+        # aren't identical, then don't try to be clever with GitDependenciesNotReachable
+        # but instead raise DependencyFileNotResolvable and report the whole error.
+        # This would have resulted in a more obvious error message for #4625
```

Comment:
I actually wasn't 100% sure on this... I could see a case where a multi-line error string may have multiple URLs, and the last one is indeed the proper one to match on... But given that `GitDependenciesNotReachable` tells the user that Dependabot can't reach the repo and to update their access creds, that's a fairly opinionated solution... IMO I'd only want to report that if 110% sure that's the issue, otherwise surface the whole error message (I assume that's what `DependencyFileNotResolvable` does... if it just hides it in logs that only Dependabot devs can see, then that's not so helpful either.)

Reply:
I see these cases as most useful to raise `GitDependenciesNotReachable`.

1. You have a direct dependency on a private github repo but Dependabot doesn't have access to it.
2. You have an indirect dependency on a github private repo (from another direct private github dependency) but Dependabot doesn't have access to it.

Do we have examples of what multiple URLs look like in this error message handy? I imagine this could get tricky if you had a failing indirect dependency on another host that came from one of your direct github dependencies.